src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java

```diff
@@ -832,8 +832,16 @@ public void initCoverage(CoverageInfo cinfo, final String coverageName) throws E
 
         CoordinateReferenceSystem nativeCRS = cinfo.getNativeCRS();
 
-        if (cinfo.getSRS() == null) {
-            cinfo.setSRS(nativeCRS.getIdentifiers().toArray()[0].toString());
+        if (nativeCRS != null) {
+            try {
+                Integer code = CRS.lookupEpsgCode(nativeCRS, false);
+                if (code != null) {
+                    cinfo.setSRS("EPSG:" + code);
+                    cinfo.setProjectionPolicy(ProjectionPolicy.REPROJECT_TO_DECLARED);
```

Comment:
If a match was found, best to set the policy to "force declared". If there is a match, it means no reprojection is needed.

Reply:
I was going by this [note](https://github.com/geoserver/geoserver/blob/bbf9b4f4efadf38a3d02bc10028077e34157a9a1/src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java#L987)  in the buildCoverageInternal method that the UI uses. Should that also be FORCE_DECLARED?